src/atmos_param/socrates/interface/socrates_calc.F90

```diff
@@ -0,0 +1,206 @@
+real(r_def), intent(in) :: h2o(n_profile, n_layer)
+!   Mass mixing ratio of water vapour
+real(r_def), intent(in) :: o3(n_profile, n_layer)
+!   Mass mixing ratio of ozone
+real(r_def), intent(in) :: co2(n_profile, n_layer)
+!   Mass mixing ratio of carbon dioxide
+
+real(r_def), intent(in) :: t_rad_surf(n_profile)
+!   Effective radiative temperature over whole grid-box
+real(r_def), intent(in) :: cos_zenith_angle(n_profile)
+!   Cosine of solar zenith angle
+real(r_def), intent(in) :: solar_irrad(n_profile)
+!   Solar irradiance at top-of-atmosphere (mean over timestep)
+real(r_def), intent(in) :: orog_corr(n_profile)
+!   Orographic correction factor
+
+logical, intent(in) :: l_planet_grey_surface
+!   Set a single grey albedo / emissivity for the surface
+real(r_def), intent(in) :: planet_albedo(:)
+!   Surface albedo used for SW calculations
+real(r_def), intent(in) :: planet_emissivity
+!   Surface emissivity used for LW calculations
+
+real(r_def), intent(in) :: layer_heat_capacity(n_profile, n_layer)
+!   Heat capacity of layer
+
+real(r_def), intent(out) :: flux_direct(n_profile, 0:n_layer)
+!   Direct (unscattered) downwards flux (Wm-2)
+real(r_def), intent(out) :: flux_down(n_profile, 0:n_layer)
+!   Downwards flux (Wm-2)
+real(r_def), intent(out) :: flux_up(n_profile, 0:n_layer)
+!   Upwards flux (Wm-2)
+real(r_def), intent(out) :: heating_rate(n_profile, n_layer)
+!   Heating rate (Ks-1)
+
+REAL(r_def), INTENT(inout), optional :: spectral_olr(:,:)
+!   Spectral OLR
+
+
+! Dimensions:
+TYPE (StrDim) :: dimen
+
+! Atmospheric properties:
+TYPE(StrAtm) :: atm
+
+! Boundary conditions:
+TYPE(StrBound) :: bound
+
+! Cloud properties:
+TYPE(StrCld) :: cld
+
+! Aerosol properties:
+TYPE(StrAer) :: aer
+
+! Output fields from core radiation code:
+TYPE(StrOut) :: radout
+
+integer(i_def) :: l, i
+!   Loop variablesi
+
+!DIAG Diagnostic
+logical :: used
+
+
+
+call set_control(control)
+
+call set_dimen(control, dimen, spectrum, n_profile, n_layer,                   &
+  n_cloud_layer, n_aer_mode, cld_subcol_gen, cld_subcol_req)
+
+call set_atm(control, dimen, spectrum, atm, n_profile, n_layer,                &
+  p_layer, t_layer, t_layer_boundaries, d_mass, density, h2o, o3, co2)
+
+call set_bound(control, dimen, spectrum, bound, n_profile,                     &
+  t_rad_surf, cos_zenith_angle, solar_irrad, orog_corr,                        &
+  l_planet_grey_surface, planet_albedo, planet_emissivity)
+
+call set_cld(control, dimen, spectrum, cld, n_profile)
+
+call set_aer(control, dimen, spectrum, aer, n_profile)
+
+! DEPENDS ON: radiance_calc
+call radiance_calc(control, dimen, spectrum, atm, cld, aer, bound, radout)
+
+
+! set heating rates and diagnostics
+do l=1, n_profile
+  do i=1, n_layer
+    heating_rate(l, i) = (radout%flux_down(l,i-1,1)-radout%flux_down(l,i,1)    &
+                        + radout%flux_up(l,i,1)-radout%flux_up(l,i-1,1))       &
+                       / layer_heat_capacity(l, i)
+  end do
+end do
+
```

Comment:
Just been checking where `heating_rate` is calculated, and it comes from here, where the net flux is divided by the layer heat capacity. The units of this are K/s, so `heating_rate` is indeed a temperature tendency. Will correct the output units accordingly.